### DISCUSSION TITLE
Pushing nix-on-droid derivations to a cache ?

### DISCUSSION THREAD

**Discussion by 573** [nix-community/nix-on-droid#176] [Q&A] [answered]
I plan to push the derivations built by `nix-on-droid switch` to my cachix to shorten further builds.

The author of https://gvolpe.github.io/blog/nixos-binary-cache-ci/ has worked this out nicely.

Now my question would be:

Given the default.nix would read:  

```nix
{ pkgs ? import <nixpkgs> {
    config = { allowUnfree = true; };
  }
}:

{
  home = pkgs.callPackage ./default-hm.nix {};
  system = pkgs.callPackage ./default-nix-on-droid.nix {};
}
```

What would the default-nix-on-droid.nix be like ?

In other terms what came where the line reads `pkgs.nixos ...` in:

```nix
{ pkgs }:

{
  system = pkgs.lib.recurseIntoAttrs (
    pkgs.nixos [ ./nix-on-droid.nix ]
  );

  recurseForDerivations = true;
}
```

Thank you so much for this awesome project !

**Comment by Gerschtli** [comment, marked as answer]
You need to build `activationPackage` in `<nix-on-droid/modules>`. See https://github.com/t184256/nix-on-droid/blob/master/nix-on-droid/nix-on-droid.sh#L26.

So something like that:

```nix
{
  system = (import <nix-on-droid/modules> {}).activationPackage;
}
```

See https://github.com/t184256/nix-on-droid/blob/master/modules/default.nix for arguments to `<nix-on-droid/modules>`.

**Reply by 573** [reply]
Using the default location of nix-on-droid.nix this just works.
